csrc/scheduler/registry.cpp

```diff
@@ -1504,6 +1523,15 @@ class ReductionScheduler : public SchedulerEntry {
       Fusion* fusion,
       SchedulerRuntimeInfo& runtime_info,
       HeuristicSummary* data_cache = nullptr) {
+    for (auto rop : ir_utils::getReductionOps(fusion)) {
+      // Reject if there are any zero-size reductions, since these would prefer
+      // pointwise, or if there are reductions with zero-size concrete
+      // dimensions, since these would prefer NoOp
+      if (ir_utils::isReductionOverSizeZero(rop, runtime_info) ||
```

Comment:
What would happen if there are two reduction domains and only one of them is size-0? It doesn't seem like a full op as there's still a non-size-0 reduction domain, but `isReductionOverSizeZero` would return `true`.

Reply:
My understanding may be flawed, but in that case there are still zero elements being reduced aren't there? For example we could merge all the reduction IterDomains and would wind up with an extent-zero reduction domain.

Reply:
I agree with this interpretation.